spec/lib/reporting_spec.rb

```diff
@@ -2,102 +2,114 @@
+          expect(note_1.body).to eq('test note body')
+          expect(note_2.body).to eq('test note body 2')
+          expect(user1.notes.count).to eq(2)
+          expect(with_notes).to eq(1)
+          expect(with_notes_percentage).to eq(100.0)
+          expect(without_notes).to be_zero
+          expect(without_notes_percentage).to be_zero
+        end
       end
-    end
 
-    context 'when there is 1 user with multiple notes' do
-      it 'notes count is 1 and percentage is 100' do
-        expect(user1_note1.body).to eq('test note body')
-        expect(user1_note2.body).to eq('test note body 2')
-        expect(user1.notes.count).to eq(2)
-        expect(reporting.users[:with_notes]).to eq(1)
-        expect(reporting.users[:with_notes_percentage]).to eq(100.0)
+      context 'when half of users have notes' do
+        it 'percentage is 50' do
+          expect(note_1.body).to eq('test note body')
+          expect(user2.email).to eq('test2@example.com')
+          expect(user1.notes.count).to eq(1)
+          expect(user2.notes.count).to be_zero
+          expect(with_notes).to eq(1)
+          expect(with_notes_percentage).to eq(50.0)
+          expect(without_notes).to eq(1)
+          expect(without_notes_percentage).to eq(50.0)
+        end
       end
-    end
 
-    context 'when 1 user has notes and 1 has none' do
-      it 'notes count is 1 and percentage is 50' do
-        expect(user1_note1.body).to eq('test note body')
-        expect(user2.email).to eq('test2@example.com')
-        expect(user1.notes.count).to eq(1)
-        expect(user2.notes.count).to eq(0)
-        expect(reporting.users[:with_notes]).to eq(1)
-        expect(reporting.users[:with_notes_percentage]).to eq(50.0)
+      context 'when 2 users have notes' do
+        it 'notes count is 2 and percentage is 100' do
+          expect(note_1.body).to eq('test note body')
+          expect(note_3.body).to eq('test note body')
+          expect(user1.notes.count).to eq(1)
+          expect(user2.notes.count).to eq(1)
+          expect(with_notes).to eq(2)
+          expect(with_notes_percentage).to eq(100.0)
+          expect(without_notes).to be_zero
+          expect(without_notes_percentage).to be_zero
+        end
       end
-    end
 
-    context 'when 2 users have notes' do
-      it 'notes count is 2 and percentage is 100' do
-        expect(user1_note1.body).to eq('test note body')
-        expect(user2_note.body).to eq('test note body')
-        expect(user1.notes.count).to eq(1)
-        expect(user2.notes.count).to eq(1)
-        expect(reporting.users[:with_notes]).to eq(2)
-        expect(reporting.users[:with_notes_percentage]).to eq(100.0)
+      context 'when a user is closed' do
+        it 'their notes are deleted' do
+          expect(note_1.body).to eq('test note body')
+          expect(user1.email).to eq('test1@example.com')
+          expect(user1.notes.count).to eq(1)
+          user1.redact!
```

Comment:
What can make specs better is to put the prerequisites in a `before` block. So **given** a user exists, **when** they close the account, **then** we get...

- **Given** describes the lazy `let`s in the `describe` block
- **When** describes the `context` specific `before` block
- **Then** tests assertions that are relevant

In this case, the assertions that notes are deleted is the responsibility of a `User#redact!` spec. What this test should do is assert that our stats remain unchanged and are not counting closed accounts.